test/functional/rpc_getblockfrompeer.py

```diff
@@ -13,14 +13,19 @@
 
 class GetBlockFromPeerTest(BitcoinTestFramework):
     def set_test_params(self):
-        self.num_nodes = 2
+        self.num_nodes = 3
+        self.extra_args = [
+            [],
+            [],
+            ["-fastprune", "-prune=1"]
+        ]
 
     def setup_network(self):
         self.setup_nodes()
 
-    def check_for_block(self, hash):
+    def check_for_block(self, node, hash):
         try:
-            self.nodes[0].getblock(hash)
+            node.getblock(hash)
```

Comment:
Maybe faster/cleaner to pass only the node number instead of whole node objects (also, use named args)
```diff
     def check_for_block(self, node, hash):
         try:
-            self.nodes[0].getblock(hash)
+            self.nodes[node].getblock(hash)
             return True
         except JSONRPCException:
             return False
@@ -65,7 +65,7 @@ class GetBlockFromPeerTest(BitcoinTestFramework):
 
         self.log.info("Successful fetch")
         result = self.nodes[0].getblockfrompeer(short_tip, peer_0_peer_1_id)
-        self.wait_until(lambda: self.check_for_block(self.nodes[0], short_tip), timeout=1)
+        self.wait_until(lambda: self.check_for_block(node=0, hash=short_tip), timeout=1)
         assert_equal(result, {})
 
         self.log.info("Don't fetch blocks we already have")
@@ -86,7 +86,7 @@ class GetBlockFromPeerTest(BitcoinTestFramework):
         assert_equal(len(peers), 1)
         peer_2_peer_0_id = peers[0]["id"]
         result = self.nodes[2].getblockfrompeer(pruned_block, peer_2_peer_0_id)
-        self.wait_until(lambda: self.check_for_block(self.nodes[2], pruned_block), timeout=1)
+        self.wait_until(lambda: self.check_for_block(node=2, hash=pruned_block), timeout=1)
         assert_equal(result, {})

```

Reply:
done